examples/multi_gpu/papers100m_multigpu.py

```diff
@@ -0,0 +1,137 @@
+import argparse
+import os
+import time
+
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.nn.functional as F
+from ogb.nodeproppred import PygNodePropPredDataset
+from torch.nn.parallel import DistributedDataParallel
+from torchmetrics import Accuracy
+
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import GCNConv
+
+
+def pyg_num_work():
+    num_work = None
+    if hasattr(os, "sched_getaffinity"):
+        try:
+            num_work = len(os.sched_getaffinity(0)) / 2
```

Comment:
This might open too many processes, leading to slowdown, as there will be `(pyg_num_work()+1)*torch.cuda.device_count()` processes on a single machine, but users are expected to adjust this anyway, so it may be fine as is.

Reply:
yeah it might need tuning, it seemed relatively performant for my set up